### PR TITLE
Move repos to use spacewalk version 2.3 and using python 2.7

### PR DESCRIPTION
**Pull request by briandoodyie**

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -20,9 +20,9 @@
 case node['platform_family']
 when 'rhel'
   platform_major = node['platform_version'][0]
-  default['spacewalk']['server']['repo_url'] = "http://spacewalk.redhat.com/yum/2.2/RHEL/#{platform_major}/#{arch}/spacewalk-repo-2.2-1.el#{platform_major}.noarch.rpm"
+  default['spacewalk']['server']['repo_url'] = "http://spacewalk.redhat.com/yum/2.3/RHEL/#{platform_major}/#{arch}/spacewalk-repo-2.3-4.el#{platform_major}.noarch.rpm"
 when 'fedora'
-  default['spacewalk']['server']['repo_url'] = "http://spacewalk.redhat.com/yum/2.2/Fedora/#{node['platform_version']}/#{arch}/spacewalk-repo-2.2-1.fc#{node['platform_version']}.noarch.rpm"
+  default['spacewalk']['server']['repo_url'] = "http://spacewalk.redhat.com/yum/2.3/Fedora/#{node['platform_version']}/#{arch}/spacewalk-repo-2.3-4.fc#{node['platform_version']}.noarch.rpm"
```
Comment (philicious) [inline attributes/default.rb:25]: when updating the default attributes, I'd rather go to 2.4 as thats latest greatest. 
For other versions people can still override this attrib.

(I didnt have the chance yet to upgrade my own Spacewalk or testing an upgrade done via Chef)

**Comment by briandoodyie** [comment]
Hi Philicious, I'm still working through versions. Not intentional to have this pull request against your master yet, just out own.. Still a lot of cleanup and testing needed. What we are trying to do is implement the redhat server piece. A lot done but haven't yet figured out how to leverage your debFile piece. Will keep work to a branch on forked repo for now

**Comment by briandoodyie** [comment]
Cant remember the exact bug the fix was for but it was just not installing spacewalk on centos until we upgraded the version of spacewalk. I only remember it was some xmlrpc failure registereing clients and was fixed in the spacewalk 2.3

**Comment by philicious** [comment]
@briandoodyie just lemme know if you need any help/advice. I've done alot with SW over past years with both rhel/centos/ubuntu clients.

I'm not sure what you mean with "implement the redhat server piece", are you refering to setting up reposync+errata = what ::ubuntu recipe does for ubuntu clients?

The ::default recipe should work for doing a basic installation tho. Its tested at least on CentOS6.6 

The debfile part https://github.com/philicious/spacewalk-server-chef/blob/master/recipes/ubuntu.rb#L16-L22 adds the missing LZMA compression required for handling of some deb packages. All it does is fix this line `PART_EXTS = ['gz', 'bz2', 'xz', 'lzma']`

If you provide more feedback like error logs or whatever, I might be able to help you way better.

**Comment by philicious** [comment]
obsoleted by #6

**Pull request closed without merging**